test/app-tap/gh-2717-no-quit-sigint.test.lua

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env tarantool
+
+local fiber = require('fiber')
+local tap = require('tap')
+local popen = require('popen')
+
+--
+-- gh-2717: tarantool console quit on sigint
+--
+
+local TARANTOOL_PATH = arg[-1]
+
+local res = tap.test('gh-2717', function(test)
+    print(TARANTOOL_PATH)
+
+    test:plan(2)
+    local prompt = 'tarantool> '
+    local ph = popen.shell(TARANTOOL_PATH .. ' -i', 'r')
+    assert(ph.pid, 'popen error while executing ' .. TARANTOOL_PATH)
+
+    fiber.sleep(0.5)
+    ph:signal(popen.signal.SIGINT)
+    fiber.sleep(0.5)
```

Comment:
General principle: wait for an event instead of using constant sleep.

If you need to wait until a process will die — check its state in a loop (till some timeout).

Reply:
In this case I'm not waiting until the process will die. There is some another problem with this test, specifically I notice that if I'm starting Tarantool in interactive mode and fork it per `open.shell(...)` it needs some time until it will be able to fetch the prompt from STDOUT of shell. Otherwise if I call `ph:read()` without any timeout it gets the either empty output or Tarantool version-help output (every time is different) so that breaks the test check. I have simplified the test, so that just checks if the process is still living after SIGINT. Anyway with next update I'm going to come up with something in described above case. Now it looks like I'm trying to call `ph:read()` until the prompt is read but it breaks always with some strange output.

Reply:
Okay, leave me know, when you'll resolve all those problems with the test.

Reply:
I have resolved this case by waiting until the shell prints the output that is needed.

Reply:
Okay so.